lib/dpl/provider/pages.rb

```diff
@@ -0,0 +1,76 @@
+module DPL
+  class Provider
+    class Pages < Provider
+      """Implements Github Pages deployment
+
+      Options:
+        - repo [optional, for pushed to other repos]
+        - github-token [required]
+        - target-branch [optional, defaults to gh-pages]
+        - local-dir [optional, defaults to `pwd`]
+        - fqdn [optional]
+        - project-name [optional, defaults to fqdn or repo slug]
+        - email [optional]
+        - name [optional]
+      """
+
+      require 'tmpdir'
+
+
+      def initialize(context, options)
+        super
+
+        @build_dir = options[:local_dir] || '.'
+
+        @project_name = options[:project_name] || fqdn || slug
+        @gh_fqdn = fqdn
+
+        @gh_ref = "github.com/#{slug}.git"
+        @target_branch = options[:target_branch] || 'gh-pages'
+        @gh_token = option(:github_token)
+
+        @gh_email = options[:email] || 'deploy@travis-ci.org'
```

Comment:
I don't have a better idea. I guess I'm OK with documenting these default values in README, then.